hooks/tk-multi-publish2/basic/publish_lookfiles.py

```diff
@@ -0,0 +1,426 @@
+        """
+        item_settings = self._get_item_settings(settings)
+        # Check the templates exist
+        work_template = item.properties["work_template"]
+        if not work_template:
+            raise Exception("'{}': Work template '{}' doesn't exist".format(item.name, work_template.name))
+        publish_template = item.properties["publish_template"]
+        if not publish_template:
+            raise Exception("'{}': Publish template '{}' doesn't exist".format(item.name, publish_template.name))
+        # Check the file exists still. It might have been removed from the filesystem after collection
+        path = item_settings["to_publish"]
+        if not os.path.exists(path):
+            raise Exception("The file '{}' no longer exists on disk. Has it been deleted/published?".format(path))
+        # Check the filepath is valid
+        if not work_template.validate(path):
+            raise Exception("The filepath '{}' does not match the template '{}'".format(path, work_template.name))
+        # Check if file has already been copied to the publish location
+        fields = work_template.validate_and_get_fields(path)
+        publish_path = publish_template.apply_fields(fields)
+        item.properties["publish_path"] = publish_path
+        item.properties["path"] = path
+        if os.path.exists(publish_path):
+            raise Exception("The file '{}' has already been copied to the publish location.".format(path))
+
+        return True
+
+    def publish(self, settings, item):
+        """
+        Executes the publish logic for the given item and settings.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+        publish_path = item.properties["publish_path"]
+        publish_dir = os.path.dirname(publish_path)
+
+        self.logger.debug("Making directory: '{}'".format(publish_dir))
+        engine = sgtk.platform.current_engine()
+        engine.ensure_folder_exists(publish_dir)
+
+        item.properties["publish_type"] = "Katana Look File"
+
+        super(KatanaLookFilePublishPlugin, self).publish(settings, item)
+
+    def create_settings_widget(self, parent):
+        """
+        Creates a Qt widget, for the supplied parent widget (a container widget
+        on the right side of the publish UI).
+
+        :param parent: The parent to use for the widget being created
+        :return: A QtGui.QWidget or subclass that displays information about
+            the plugin and/or editable widgets for modifying the plugin's
+            settings.
+        """
+        description_widget = super(KatanaLookFilePublishPlugin, self).create_settings_widget(None)
+        return LookFilePublishWidget(parent=parent, description_widget=description_widget) 
+
+    def get_ui_settings(self, widget):
```

Comment:
Decorate it as `@staticmethod`?

```suggestion
    @staticmethod
    def get_ui_settings(widget):
```

Reply:
this is an overridden method